pyadjoint/optimization/rol_solver_checkpointed.py

```diff
@@ -0,0 +1,283 @@
+"""
+    Checkpointed optimisation with ROL2.0
+"""
+import os
+import shutil
+
+from ..tape import no_annotations
+from mpi4py import MPI
+from .rol_solver import (
+    ROLVector, ROLObjective, ROLSolver)
+from .optimization_solver import OptimizationSolver
+from firedrake.checkpointing import CheckpointFile
+from firedrake import utils
+
+vector_registry = []
```

Comment:
Might be worth actually documenting how this process works, and particularly the vector registry?